cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -324,21 +324,19 @@ func (h *bundleHandler) resolveCharmsAndEndpoints() error {
 
 		// To deploy by revision, the revision number must be in the origin for a
 		// charmhub charm.
-		if charm.CharmHub.Matches(ch.Schema) {
```

Comment:
By this point we know the charm is not a local charm, so must be a charmhub charm

Reply:
Yes, this would have been left over from when we had both cs and ch charms